index.js

```diff
@@ -25,63 +24,49 @@ function search(buf) {
 	}
 
 	return false;
-}
-
-exports.buffer = function (buf) {
-	return search(buf);
 };
 
-exports.stream = function (cb) {
-	var prevLastByte = new Buffer(1);
-	var searchDone = false;
+exports.buffer = search;
 
-	return through(function (data, enc, cb2) {
-		if (searchDone) {
-			cb(null, data);
-			return;
-		}
+exports.stream = stream => {
+	return new Promise((resolve, reject) => {
+		let prevLastByte = new Buffer(1);
+		let searchDone = false;
 
-		prevLastByte = new Buffer(data[data.length - 1]);
+		stream.on('data', data => {
+			prevLastByte = new Buffer(data[data.length - 1]);
 
-		var res = exports.buffer(Buffer.concat([prevLastByte, data]), true);
+			if (search(Buffer.concat([prevLastByte, data]))) {
+				searchDone = true;
+				stream.destroy();
+			}
+		});
 
-		if (res === true) {
-			searchDone = true;
-			cb(true);
-		}
+		stream.on('error', reject);
 
-		cb2(null, data);
-	}, function (cb2) {
-		if (!searchDone) {
-			cb(false);
-		}
+		stream.on('end', () => {
+			stream.destroy();
```

Comment:
You can remove the listener:

```js
stream.removeListener('end', fn);
```

You need to move the listener function into a standalone function so it can be referenced in both the subscription and unsubscription.